core/server/common/src/test/java/alluxio/master/MasterRegistryTest.java

```diff
@@ -0,0 +1,215 @@
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+import alluxio.exception.ExceptionMessage;
+import alluxio.master.journal.JournalInputStream;
+import alluxio.master.journal.JournalOutputStream;
+import alluxio.proto.journal.Journal;
+
+import org.apache.thrift.TProcessor;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class MasterRegistryTest {
+
+  public abstract class TestMaster implements Master {
+    @Override
+    public Map<String, TProcessor> getServices() {
+      return null;
+    }
+
+    @Override
+    public void processJournalCheckpoint(JournalInputStream inputStream) throws IOException {
+      return;
+    }
+
+    @Override
+    public void processJournalEntry(Journal.JournalEntry entry) throws IOException {
+      return;
+    }
+
+    @Override
+    public void start(boolean isLeader) throws IOException {
+      return;
+    }
+
+    @Override
+    public void stop() throws IOException {
+      return;
+    }
+
+    @Override
+    public void streamToJournalCheckpoint(JournalOutputStream outputStream) throws IOException {
+      return;
+    }
+
+    @Override
+    public void transitionToLeader() {
+      return;
+    }
+  }
+
+  public class MasterA extends TestMaster {
+    @Override
+    public String getName() {
+      return "A";
+    }
+
+    @Override
+    public Set<Class<?>> getDependencies() {
+      Set<Class<?>> deps = new HashSet<>();
+      deps.add(MasterB.class);
+      return deps;
+    }
+  }
+
+  public class MasterB extends TestMaster {
+    @Override
+    public String getName() {
+      return "B";
+    }
+
+    @Override
+    public Set<Class<?>> getDependencies() {
+      Set<Class<?>> deps = new HashSet<>();
+      deps.add(MasterC.class);
+      return deps;
+    }
+  }
+
+  public class MasterC extends TestMaster {
+    @Override
+    public String getName() {
+      return "C";
+    }
+
+    @Override
+    public Set<Class<?>> getDependencies() {
+      Set<Class<?>> deps = new HashSet<>();
+      deps.add(MasterD.class);
+      return deps;
+    }
+  }
+
+  public class MasterD extends TestMaster {
+    @Override
+    public String getName() {
+      return "C";
+    }
+
+    @Override
+    public Set<Class<?>> getDependencies() {
+      Set<Class<?>> deps = new HashSet<>();
+      deps.add(MasterA.class);
+      return deps;
+    }
+  }
+
+  @Test
+  public void registry() {
+    String[] expectedNames = {"C", "B", "A"};
+
+    {
+      MasterRegistry registry = new MasterRegistry();
+      registry.add(MasterA.class, new MasterA());
+      registry.add(MasterB.class, new MasterB());
+      registry.add(MasterC.class, new MasterC());
+
+      int i = 0;
+      for (Master master : registry.getMasters()) {
```

Comment:
if we make `expectedNames` and the return value of `getMasters` lists, we can use `.equals` instead of iterating.